### ISSUE TITLE
Parsing rules are somewhat unexpected

### ISSUE THREAD

**Issue by x-yuri** [theskumar/python-dotenv#276]
`A=1 2` (one space) works, but `A=1  2` (two spaces) produces an error. I've run into it with `docker-compose`:

`.env`:

```yaml
A=1  2
```

`docker-compose.yml`:

```yaml
version: '3'
services:
  app:
    image: alpine:3.12
    command: /app/1.sh
    env_file: .env
    volumes:
      - .:/app
```

`1.sh`:

```sh
#!/bin/sh -eux
set | grep ^A
```

```
$ docker-compose up -d && docker-compose logs -f
WARNING: Python-dotenv could not parse statement starting at line 1
WARNING: Python-dotenv could not parse statement starting at line 1
WARNING: Python-dotenv could not parse statement starting at line 1
WARNING: Python-dotenv could not parse statement starting at line 1
Creating network "overriding-vars_default" with the default driver
Creating overriding-vars_app_1 ... done
WARNING: Python-dotenv could not parse statement starting at line 1
WARNING: Python-dotenv could not parse statement starting at line 1
Attaching to overriding-vars_app_1
app_1  | + set
app_1  | + grep ^A
overriding-vars_app_1 exited with code 1
```

With `A=1 2`:

```
$ docker-compose up -d && docker-compose logs -f
Recreating overriding-vars_app_1 ... done
Attaching to overriding-vars_app_1
app_1  | + set
app_1  | + grep ^A
app_1  | A='1 2'
overriding-vars_app_1 exited with code 0
```

I'll provide a Python test case, if need be.

https://github.com/theskumar/python-dotenv/blob/v0.14.0/src/dotenv/parser.py#L175-L176

**Comment by bbc2** [comment]
That's very interesting. Thank you for reporting. I'll look into it.

**Comment by bbc2** [comment]
I think I've found a fix for the bug you reported.  Could you let me know if the fix in #277 works for you?
